File: 1765-merge-in-between-linked-lists/merge-in-between-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeInBetween(ListNode* list1, int a, int b, ListNode* list2) {
        ListNode* prev=list1;

        //prev will point in node one before a
        for(int i=0;i<a-1;i++)
        {
            prev=prev->next;
        }

        ListNode* track=prev;

        for(int i=a;i<=b;i++)
        {
            prev=prev->next;
        }

        //traversing list2
        track->next=list2;
        ListNode* end=list2;
        while(end->next!=NULL)
        {
            end=end->next;
        }

        //connnect end to prev of list1
        end->next=prev->next;

        return list1;
    }
};

File: 1765-merge-in-between-linked-lists/merge-in-between-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        prev = list1

        # prev will point to the node one before index 'a'
        for i in range(a - 1):
            prev = prev.next

        track = prev

        # Move prev to the node at index 'b'
        for i in range(a, b + 1):
            prev = prev.next

        # Connect the end of list2 to the node after index 'b'
        track.next = list2
        end = list2
        while end.next is not None:
            end = end.next

        end.next = prev.next  # Connect end of list2 to the node after index 'b'

        return list1